WordPress/gutenberg
# Patterns: remove "Manage all parts" page & link

Part https://github.com/WordPress/gutenberg/issues/59659 and https://github.com/WordPress/gutenberg/issues/55083

## What?

- Removes the link to "Manage all template parts" in patterns.
- The `/wp_template_part/all` URL is still accessible to hybrid themes (classic themes that declare support for `block-template-part`).
- The `/patterns` URL is still accessible unofficially to classic themes (with or without support for `block-template-part`).

| | Before | After |
| --- | --- | --- |
| **Block themes**. The "Manage all parts" link is no longer accessible. | <img width="1648" alt="Captura de ecrã 2024-04-22, às 12 47 32" src="https://github.com/WordPress/gutenberg/assets/583546/8b4091ab-9e34-4c03-9660-ea26b375683c"> | <img width="1648" alt="Captura de ecrã 2024-04-22, às 12 55 08" src="https://github.com/WordPress/gutenberg/assets/583546/e33038ed-9bef-4d66-879f-f79eea34fa4f"> |
| **Hybrid themes**. They still can access the `/wp_template_part/all` URL, as before | <img width="1648" alt="Captura de ecrã 2024-04-22, às 12 47 49" src="https://github.com/WordPress/gutenberg/assets/583546/df9a98b9-84d4-4ff7-89b7-a3dcfde665cd"> | <img width="1648" alt="Captura de ecrã 2024-04-22, às 13 10 32" src="https://github.com/WordPress/gutenberg/assets/583546/5fa0bd45-fe9d-4fba-a59f-da5cbbadb63b"> |
| **Hybrid themes**. They still can access the unofficially supported `/patterns` URL, as before | <img width="1648" alt="Captura de ecrã 2024-04-22, às 12 48 15" src="https://github.com/WordPress/gutenberg/assets/583546/20f3a927-efb5-4cb0-ba79-2d5e3fafb661"> | <img width="1648" alt="Captura de ecrã 2024-04-22, às 13 10 39" src="https://github.com/WordPress/gutenberg/assets/583546/1350a2e8-b314-41a6-b72e-0c2d566ec877"> |
| **Classic themes**. They still can access the unofficially supported `/patterns` URL, as before | <img width="1648" alt="Captura de ecrã 2024-04-22, às 12 48 33" src="https://github.com/WordPress/gutenberg/assets/583546/0c6f7b41-ab89-458b-9008-8680d08ef866"> | <img width="1648" alt="Captura de ecrã 2024-04-22, às 13 00 00" src="https://github.com/WordPress/gutenberg/assets/583546/47457edd-7164-4a2d-ab04-0d72514772a8"> |


## Why?

- That's the target design we want, as per https://github.com/WordPress/gutenberg/issues/59659
- Over small PRs, all functionality have been migrated to the existing Parts pages in Patterns, so this is just removing the link and the code.

## How?

> The `/wp_template_part/all` URL still exists, because hybrid themes require it.

This PR updates the components that are used for that route. It now loads the code from Patterns (sidebar & content components). The reason for this is that in a future step hybrid themes will also have access to the Patterns from the site editor, so we'll load the same components. 

Changes:

- Router: `packages/edit-site/src/components/layout/router.js`
  - Updates the components used. From `PageTemplatesTemplateParts` to `PagePatterns` and from `SidebarNavigationScreenTemplatesBrowse` to `SidebarNavigationScreenPatterns`.
- Patterns
  - Sidebar: `packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js`. Configures the component depending on whether we're loading the `/wp_template_part/all` URL (display only Template parts) or `/patterns` (display everything, Patterns & Parts).
  - Content: `packages/edit-site/src/components/page-patterns/index.js`. Configures the component depending on whether we're loading the `/wp_template_part/all` URL (display only Template parts) or `/patterns` (display everything, Patterns & Parts).
- Pattern (details view)
  - Sidebar: `packages/edit-site/src/components/sidebar-navigation-screen-pattern/index.js`. Updates the logic so that is goes back to `/wp_template_part/all` if the theme is not a block theme.

> Remove no longer necessary code.

The "Manage template parts" screen used different code to render the Template parts than the "Patterns" screen uses. This PR removes all of that.

Changes:

- Sidebar (`packages/edit-site/src/components/sidebar-navigation-screen-templates-browse/`):
  - `index.js`: removes all the code related to parts.
- Content (`packages/edit-site/src/components/page-templates/`):
  - Renames the directory from `packages/edit-site/src/components/page-templates-template-parts/` to `packages/edit-site/src/components/page-templates/`.
  - Renames the `PageTemplatesTemplateParts` component to `PageTemplates`.
  - Removes `packages/edit-site/src/components/page-templates-template-parts/add-new-template-part.js`, as we already have the same code in the Patterns screen.
  - `index.js`: removes all the code that was used for template parts.

## Testing Instructions

Using a block theme:

- Visit the Patterns page and verify that the link is not there.
- Verify the link back from details takes you back to the patterns page, whether you are editing a Part or a Pattern.

Using a classic theme (for example, visit `localhost:8889` and select TwentyTwentyOne):

- Visit the link `http://localhost:8889/wp-admin/site-editor.php?path=%2Fpatterns` and verify that it works as described above. It's an unofficial link, but people expect it to work.

Using a hybrid theme (for example, visit `localhost:8889` and select EmptyHybrid):

- Go to Appareance > Template parts and verify the site-editor Template Parts page is loaded and works as described above.

- Visit the link `http://localhost:8889/wp-admin/site-editor.php?path=%2Fpatterns` and verify that it works as described above. It's an unofficial link, but people expect it to work.
